### PR TITLE
Update contrib.named examples

### PR DESCRIPTION
**Pull request by fritzo**
Addresses #930

### CODE REVIEW

File: examples/contrib/named/tree_data.py [hunk 1]
```diff
@@ -87,10 +91,12 @@ def main(args):
...
 
     print('Step\tLoss')
+    loss = 0.0
     for step in range(args.num_epochs):
-        if step % 100 == 0:
-            loss = inference.step(data)
+        loss += inference.step(data)
+        if step and step % 10 == 0:
             print('{}\t{:0.5g}'.format(step, loss))
+            loss = 0.0
```
Comment (jpchen) [inline examples/contrib/named/tree_data.py]: i havent looked at this example with much detail, but why is the loss being accumulated rather than set (as it was previously)?

Comment (fritzo) [reply examples/contrib/named/tree_data.py]: I'm accumulating it until I print it, then reset the accumulator. This is to overcome noise in the loss. Previous to this PR the loss looked non-monotonic.

Comment (fritzo) [reply examples/contrib/named/tree_data.py]: (also the old logic is totally wrong: most iterations didn't even .step() 😕 )

**Comment by jpchen** [comment]
@fritzo good to merge?

**Comment by fritzo** [comment]
ready to merge
